pywebpush/__init__.py

```diff
@@ -152,31 +152,33 @@ def encode(self, data, content_encoding="aesgcm"):
             raise WebPushException("Invalid content encoding specified. "
                                    "Select from " +
                                    json.dumps(self.valid_encodings))
-        if (content_encoding == "aesgcm"):
-            salt = os.urandom(16)
         # The server key is an ephemeral ECDH key used only for this
         # transaction
         server_key = ec.generate_private_key(ec.SECP256R1, default_backend())
-        crypto_key = base64.urlsafe_b64encode(
-            server_key.public_key().public_numbers().encode_point()
-        ).strip(b'=')
+        crypto_key = server_key.public_key().public_numbers().encode_point()
+        if (content_encoding == "aesgcm"):
+            salt = os.urandom(16)
```

Comment:
Ah, that is new. It didn't before. One possible issue, though is it appears that while ece creates a salt, [it doesn't return it](https://github.com/web-push-libs/encrypted-content-encoding/blob/7af350d70954b5973f39308d9321a73f4aa9ad19/python/http_ece/__init__.py#L387) for `aesgcm`, so it's still better to create a salt here if we plan on including it in the headers.